src/protocols/bdx/tests/TestBdxTransferSession.cpp

```diff
@@ -0,0 +1,335 @@
+    senderOpts.Set(kControl_ReceiverDrive);
+    err = respondingSender.WaitForTransfer(kRole_Sender, senderOpts, transferBlockSize);
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+    err = respondingSender.HandleMessageReceived(outEvent.MsgData.Retain());
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+
+    // Responder may want to read any of the fields of the TransferInit for approval.
+    // Verify File Designator matches. Verify that the respondingSenser automatically set ReceiverDrive
+    // for the chosen ControlMode.
+    respondingSender.PollOutput(outEvent);
+    NL_TEST_ASSERT(inSuite, outEvent.EventType == TransferSession::kOutput_InitReceived);
+    NL_TEST_ASSERT(inSuite, respondingSender.GetControlMode() == kControl_ReceiverDrive);
+    NL_TEST_ASSERT(inSuite, outEvent.transferInitData.FileDesignator != nullptr);
+    NL_TEST_ASSERT(inSuite,
+                   !memcmp(testFileDes, outEvent.transferInitData.FileDesignator, outEvent.transferInitData.FileDesLength));
+
+    // Compose ReceiveAccept parameters struct and give to respondingSender
+    char testMetadata[11]      = { "hi_dad.txt" };
+    uint64_t proposedOffset    = 64;
+    uint64_t proposedLength    = 0;
+    uint16_t proposedBlockSize = transferBlockSize;
+    TransferSession::TransferAcceptData acceptData;
+    acceptData.ControlMode    = respondingSender.GetControlMode();
+    acceptData.StartOffset    = proposedOffset;
+    acceptData.Length         = 0; // Indefinite length
+    acceptData.MaxBlockSize   = proposedBlockSize;
+    acceptData.Metadata       = reinterpret_cast<uint8_t *>(testMetadata);
+    acceptData.MetadataLength = 11;
+
+    err = respondingSender.AcceptTransfer(acceptData);
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+
+    // Verify Sender emits ReceiveAccept message for sending
+    respondingSender.PollOutput(outEvent);
+    NL_TEST_ASSERT(inSuite, outEvent.EventType == TransferSession::kOutput_MsgToSend);
+    System::PacketBufferHandle acceptMsgToSend = outEvent.MsgData.Retain();
+    NL_TEST_ASSERT(inSuite, !acceptMsgToSend.IsNull());
+
+    // Pass Accept message to initiatingReceiver
+    err = initiatingReceiver.HandleMessageReceived(acceptMsgToSend.Retain());
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+
+    // Verify received ReceiveAccept.
+    // Client may want to inspect TransferControl, MaxBlockSize, StartOffset, Length, and Metadata, and may choose to reject the
+    // Transfer at this point.
+    initiatingReceiver.PollOutput(outEvent);
+    NL_TEST_ASSERT(inSuite, outEvent.EventType == TransferSession::kOutput_AcceptReceived);
+    NL_TEST_ASSERT(inSuite, outEvent.transferAcceptData.ControlMode == kControl_ReceiverDrive);
+    NL_TEST_ASSERT(inSuite, outEvent.transferAcceptData.StartOffset == proposedOffset);
+    NL_TEST_ASSERT(inSuite, outEvent.transferAcceptData.Length == proposedLength);
+    NL_TEST_ASSERT(inSuite, outEvent.transferAcceptData.MaxBlockSize == transferBlockSize);
```

Comment:
The assertion here should be that it is <= transferBlockSize and a power of two IIRC

Reply:
I actually don't see the power of 2 restriction anywhere in the spec, but I remember discussing it with you. Did we forget to add it?

Reply:
this is part of the OTA spec, not BDX, so will leave as is